test/suites/ubuntu-core-edgex-image/test.sh

```diff
@@ -1,23 +1,37 @@
-#!/bin/bash -e
+#!/bin/bash -ex
 
-ssh "$USER@localhost" -p 8022 "snap install curl"
+SSH_USER="$USER"
+SSH_PORT="8022"
 
-# check status of core services inside of the emulator
+# Install curl on host
+snap install curl
+
+# Check status of core services inside of the emulator
 ports=(59880 59881 59882)
 
 for port in "${ports[@]}"
 do
-  ssh "$USER@localhost" -p 8022 "curl -s http://localhost:$port/api/v2/ping; printf '\n'"
+  ssh "$SSH_USER@localhost" -p $SSH_PORT "curl -s http://localhost:$port/api/v2/ping"
```

Comment:
You could create an alias for `ssh "$SSH_USER@localhost" -p $SSH_PORT` to avoid repeating it and instead just call for example `remote_call "curl -s http://localhost:$port/api/v2/ping"`.